skimage/segmentation/_felzenszwalb_cy.pyx

```diff
@@ -125,3 +125,118 @@ def _felzenszwalb_grey(image, double scale=1, sigma=0.8,
         flat = flat[flat]
     flat = np.unique(flat, return_inverse=True)[1]
     return flat.reshape((height, width))
+
+
+def _felzenszwalb_rgb(image, double scale=1, sigma=0.8,
+                       Py_ssize_t min_size=20):
+    """Felzenszwalb's efficient graph based segmentation for multiple channels.
+
+    Produces an oversegmentation of a multi-channel image using a fast, minimum spanning
+    tree based clustering on the image grid.
+    The number of produced segments as well as their size can only be
+    controlled indirectly through ``scale``. Segment size within an image can
+    vary greatly depending on local contrast.
+
+    Parameters
+    ----------
+    image : ndarray
+        Input image.
+    scale : float, optional (default 1)
+        Sets the obervation level. Higher means larger clusters.
+    sigma : float, optional (default 0.8)
+        Width of Gaussian smoothing kernel used in preprocessing.
+        Larger sigma gives smother segment boundaries.
+    min_size : int, optional (default 20)
+        Minimum component size. Enforced using postprocessing.
+
+    Returns
+    -------
+    segment_mask : (height, width) ndarray
+        Integer mask indicating segment labels.
+    """
+    if image.ndim != 3:
+        raise ValueError("This algorithm works only on multi-channel 3d"
+                "images. Got image of shape %s" % str(image.shape))
+
+    image = img_as_float(image)
+
+    # rescale scale to behave like in reference implementation
+    scale = float(scale) / 255.
```

Comment:
I'm not familiar with this algorithm, but as a user point of view, presented like that, I would feel uncomfortable to pass a 16 bits image for instance :) Either we need to generalize or we should state in the docstring the limits (if it really exists/matters).